src/HTSPVFS.cpp

```diff
@@ -286,10 +228,15 @@ long long CHTSPVFS::SendFileSeek ( int64_t pos, int whence, bool force )
            m_fileId, whence, (long long)pos);
 
   /* Send */
-  if (force)
-    m = m_conn.SendAndWait0("fileSeek", m);
-  else
-    m = m_conn.SendAndWait("fileSeek", m);
+  {
+    CLockObject lock(m_conn.Mutex());
+
+    if (force)
+      m = m_conn.SendAndWait0("fileSeek", m);
+    else
+      m = m_conn.SendAndWait("fileSeek", m);
+  }
+
   if (m == NULL)
     return false;
```

Comment:
There is a mistake in a comment below this line (nto changed in the PR) it says "offset" is optional, yet requires it (which I think is correct).